# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.cpp
# Commit: 09fcee21b246
# Full Hash: 09fcee21b246dbd3622203d214234ee5116805f4
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-07-10 21:50:49
# Regressor Bug: 1561395
# File Overlap Count: 2
# Description:
#   Bug 1561395 - Move drawSnapshot API to WindowGlobalParent and allow specifying the whole viewport as a rect. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35842
# ==============================================================================

diff -r 0873b1c9bca7 -r 09fcee21b246 dom/ipc/BrowserChild.cpp
--- a/dom/ipc/BrowserChild.cpp	Thu Jul 04 00:40:02 2019 +0000
+++ b/dom/ipc/BrowserChild.cpp	Wed Jul 10 16:45:46 2019 +0000
@@ -2487,34 +2487,6 @@
   return IPC_OK();
 }
 
-mozilla::ipc::IPCResult BrowserChild::RecvRequestRootPaint(
-    const IntRect& aRect, const float& aScale, const nscolor& aBackgroundColor,
-    RequestRootPaintResolver&& aResolve) {
-  nsCOMPtr<nsIDocShell> docShell = do_GetInterface(WebNavigation());
-  if (!docShell) {
-    return IPC_OK();
-  }
-
-  aResolve(
-      gfx::PaintFragment::Record(docShell, aRect, aScale, aBackgroundColor));
-  return IPC_OK();
-}
-
-mozilla::ipc::IPCResult BrowserChild::RecvRequestSubPaint(
-    const float& aScale, const nscolor& aBackgroundColor,
-    RequestSubPaintResolver&& aResolve) {
-  nsCOMPtr<nsIDocShell> docShell = do_GetInterface(WebNavigation());
-  if (!docShell) {
-    return IPC_OK();
-  }
-
-  gfx::IntRect rect = gfx::RoundedIn(gfx::Rect(
-      0.0f, 0.0f, mUnscaledInnerSize.width, mUnscaledInnerSize.height));
-  aResolve(
-      gfx::PaintFragment::Record(docShell, rect, aScale, aBackgroundColor));
-  return IPC_OK();
-}
-
 mozilla::ipc::IPCResult BrowserChild::RecvNavigateByKey(
     const bool& aForward, const bool& aForDocumentNavigation) {
   nsIFocusManager* fm = nsFocusManager::GetFocusManager();